### PR TITLE
Add -KeyUsage that includes CertSign and change OIDs to use Extended …

### PR DESCRIPTION
**Pull request by roger-cruz**
…Key Usage rather than Application Policy

The PoSH command used here to generate a self-signed certificate, which is then promoted to a trust root CA did not work for me.
$rootCert = New-SelfSignedCertificate -CertStoreLocation cert:\CurrentUser\My -DnsName "RootCA" -TextExtension @("1.3.6.1.4.1.311.21.10={text}1.3.6.1.5.5.7.3.1,1.3.6.1.5.5.7.3.2")
Certificates signed with the above certificate will show a warning "The certificate is not valid for the selected purpose". If you try to use these certificates, you will get invalid certificate errors from clients.
The way I was able to fix this is to provide a -KeyUsage that includes "Certificate Signing" and change the OIDs to use the extended key usage rather than application policies. Once this was done for the root CA certificate, new certificates worked like a charm.
I suggest the command be changed to be like below.
$rootcert = New-SelfSignedCertificate -CertStoreLocation Cert:\CurrentUser\My -DnsName "RootCA" -TextExtension @("2.5.29.37={text}1.3.6.1.5.5.7.3.1,1.3.6.1.5.5.7.3.2") -KeyUsage CertSign,DigitalSignature,KeyEncipherment

## Summary

Describe your changes here.

Fixes #Issue_Number (if available)

### CODE REVIEW

**Comment by rpetrusha** [comment]
@Lxiamail @zhenlan @mconnew could you review this PR, please?

File: docs/framework/wcf/feature-details/how-to-create-temporary-certificates-for-use-during-development.md [hunk 1]
```diff
@@ -22,7 +22,7 @@ When developing a secure service or client using Windows Communication Foundatio
 The following command creates a self-signed certificate with a subject name of "RootCA" in the Current User Personal store.
 
 ```powershell
-PS $rootCert = New-SelfSignedCertificate -CertStoreLocation cert:\CurrentUser\My -DnsName "RootCA" -TextExtension @("1.3.6.1.4.1.311.21.10={text}1.3.6.1.5.5.7.3.1,1.3.6.1.5.5.7.3.2")
+PS $rootcert = New-SelfSignedCertificate -CertStoreLocation Cert:\CurrentUser\My -DnsName "RootCA" -TextExtension @("2.5.29.37={text}1.3.6.1.5.5.7.3.1,1.3.6.1.5.5.7.3.2") -KeyUsage CertSign,DigitalSignature,KeyEncipherment
```
Comment (bartonjs) [inline docs/framework/wcf/feature-details/how-to-create-temporary-certificates-for-use-during-development.md]: * KeyUsage should be `CertSign,CrlSign,DigitalSignature`.
* 2.5.29.37 (and its value) should be removed
* Add `2.5.29.19={text}CA=true` to TextExtension

These recommendations make a root authority look like roots under the CA/Browser Forum Baseline Requirements (v 1.6.3) (https://cabforum.org/wp-content/uploads/CA-Browser-Forum-BR-1.6.3.pdf) (section 7.1.2.1)

**Review by rpetrusha — DISMISSED** [review]
Thanks for submitting this PR, @roger-cruz. I apologize that it's taken so long for us to respond to you. Could you address the comments from @bartonjs, and then I can merge your PR.

**Comment by roger-cruz** [comment]
I’m away from a computer for the next 5 days.  Could it wait until I return from my trip and I have a chance to examine in detail?

Upon quick review, It sounds like he is adding CertSign which is what I found critical to having in the RootCA. 

Cheers

**Comment by rpetrusha** [comment]
Certainly it can wait until you have time, @roger-cruz.

**Comment by mairaw** [comment]
Any updates here?

**Comment by mairaw** [comment]
I've applied the feedback since this PR has been open for a while. @bartonjs can you make sure I captured your feedback correctly? Thanks!

**Comment by bartonjs** [comment]
Looks reasonable to me.

**Review by mairaw — APPROVED** [review]

**Pull request merged**